webrender/src/gpu_store.rs

```diff
@@ -81,8 +85,10 @@ impl<T: Clone + Default, L: GpuStoreLayout> GpuStore<T, L> {
         // Extend the data array to be a multiple of the row size.
         // This ensures memory safety when the array is passed to
         // OpenGL to upload to the GPU.
-        while items.len() % items_per_row != 0 {
-            items.push(T::default());
+        if items_per_row != 0 {
+            while items_per_row != 0 && items.len() % items_per_row != 0 {
```

Comment:
this first part of the condition is not redundant, but I don't see it as a show stopper :)